### ISSUE TITLE
failed to retrieve block from DALC with local-celestia-devnet v0.9.2 and onwards

### ISSUE THREAD

**Issue by gupadhyaya** [rollkit/rollkit#943]
```8:38PM DBG trying to retrieve block from DA daHeight=17 module=BlockManager```
```8:38PM ERR failed to retrieve block from DALC daHeight=17 errors="failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later; failed to retrieve block: current head local chain head: 16 is lower than requested height: 17 give header sync some time and retry later" module=BlockManager```

**Comment by jcstein** [comment]
same on v0.9.5?

**Comment by gupadhyaya** [comment]
yes.

**Comment by jcstein** [comment]
Should we move this to `rollkit/local-celestia-devnet`?

**Comment by gupadhyaya** [comment]
i think the fix should happen in rollkit, so the issue can be here.

**Comment by jcstein** [comment]
I wonder if this is looking for a height+1 and that's the issue?

**Comment by nashqueue** [comment]
This might have been introduced from https://github.com/celestiaorg/celestia-node/pull/2108

**Comment by jcstein** [comment]
I'm experiencing this on:
- celestia-node v0.10.0
- network: blockspacerace
- rollkit version v0.7.3

```bash
3:00PM ERR failed to retrieve block from DALC daHeight=565665 errors="failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data; failed to retrieve block: client-nd: response code is not OK: share: namespace not found in data\ngetter/ipld: failed to retrieve shares by namespace: share: namespace not found in data" module=BlockManager
```

[Full logs can be found here](https://gist.github.com/jcstein/6dbd02cbad1e01fa22bf012995e7b191)

**Comment by jcstein** [comment]
oh wow, didn't know 0.8.0 was released. nice!

**Comment by gupadhyaya** [comment]
fixed in celestia-node v0.10.3 release. i tested it using node v0.10.3 and celestia-app: v0.13.2
